migrate revisions colors to custom properties

circlecube opened the pull request:
Migrate colors for revisions to custom properties. Includes tooltip and wp-slider component. I tried to use existing color custom properties wherever it made sense.

Trac ticket: https://core.trac.wordpress.org/ticket/49930

morganisok committed:
use custom properties in the code-editor css file

circlecube committed:
migrate revisions colors to custom properties

ryelle commented on src/wp-admin/css/revisions.css:
```diff
@@ -25,8 +25,8 @@
-	background: #fff;
-	box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
+	background-color: var(--wp-admin--revisions--controls--background--pinned);
+	box-shadow: 0 1px 3px var(--wp-admin--revisions--controls--box-shadow--pinned);
```
Let's stick with the `--wp-admin--shadow` property for transparent black values, it's used with [functional rgb notation](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/rgb()) like this

```suggestion
	box-shadow: 0 1px 3px rgb(var(--wp-admin--shadow) / 0.1);
```

ryelle commented on src/wp-admin/css/revisions.css:
```diff
@@ -131,11 +131,11 @@ body.folded .revisions .loading-indicator {
 
 .revisions-controls .author-card .date {
-	color: #646970;
+	color: var(--wp-admin--revisions--author-card--date);
```
```suggestion
	color: var(--wp-admin--text-secondary--color);
```

ryelle commented on src/wp-admin/css/revisions.css:
```diff
@@ -212,10 +212,6 @@ table.diff col.ltype {
 	width: 30px;
 }
 
-table.diff tr {
-	background-color: transparent;
-}
-
```
Why was this removed?

circlecube replied:
This was removed because I couldn't see it was affecting anything, the table tr was already a transparent background. I'll revert though just in case and so that this can just focus on color updates rather than any other cleanup.

ryelle commented on src/wp-admin/css/revisions.css:
```diff
@@ -384,8 +380,8 @@ table.diff .diff-addedline ins {
 .revisions-tooltip-arrow > span {
-	border: 1px solid #dcdcde;
-	background-color: #fff;
+	border: 1px solid var(--wp-admin--revisions--tooltip--border);
+	background-color: var(--wp-admin--revisions--tooltip--background);
```
```suggestion
	border: 1px solid var(--wp-admin--surface--border-step-2);
	background-color: var(--wp-admin--surface--background);
```

ryelle reviewed (dismissed):
I think there are a lot of places here that could use existing properties, especially [the `surface` properties](https://github.com/ryelle/wordpress-develop/blob/e531b2f5dbf23aad2c5732843382b918c90984b3/src/wp-includes/css/custom-properties.css#L17-L22). Pretty much anywhere there's a white background, it can be `--wp-admin--surface--background`, most borders can use the existing border properties, etc. I suggested a few in comments here to get you started :)

circlecube committed:
use existing surface properties in place rather than adding extra vars per use.

circlecube committed:
Merge branch 'try/custom-properties' into try/custom-properties-revisions

circlecube commented:
Thanks for the feedback @ryelle - I've updated to use the existing surface properties where I could. I was able to remove a lot of the added properties in the process. I'm trying to find a balance here between adding verbose properties that inherit existing properties for each element and using existing properties in places that aren't as intuitive to me where they would be used (and I get that `intuitive` is a relative word here).

I did leave the diff values even though I used the existing notice and icon color properties where the color existed already. Not sure if we would want to drop those directly into the diff section in the revisions file though since context may be different, more clear to have a set of colors that are used and have those inherit where applicable since there's no common color list to grab from. Happy to update more if you think it needs it.

ryelle committed:
Merge branch 'wpalchemist-code-editor' into try/custom-properties

Fixes #9

ryelle committed:
Merge branch 'try/custom-properties-revisions' of https://github.com/circlecube/wordpress-develop into pr-10

ryelle committed:
Fix some naming schemes, add defaults for the slider properties

Combines sub-locations into single location tokens; and creates default `slider-handle--background` etc, inheriting from the existing properties.

ryelle committed:
Use the `--wp-admin--surface--box-shadow` property
